deployment/aws-config/config.yaml.template

```diff
@@ -1,99 +1,82 @@
-# Refinery Platform Configuration
-
-# aws-config is a directory
-# containing the configuration of
-# a refinery-platform stack.
-# The configuration is stored in two files:
-# config.yaml and,
-# auto-config.yaml
+# Refinery Platform AWS config
+
+# aws-config contains configuration for CloudFormation stacks
+# The configuration is stored in two files: config.yaml and tags.yaml
 #
-# Neither of these files exist to begin with,
-# but `config.yaml` is required.
-# Typically this file, `config.yaml.template`,
-# is copied to `config.yaml` and edited.
-
-# RDS_SNAPSHOT and DATA_SNAPSHOT are only used when
-# you are restoring a Refinery Platform Instance from
-# the snapshots of a previously created Refinery stack.
-# The first time you create a Refinery stack
-# you will not need to specify these;
-# fresh RDS instances and data volumes are created.
+# Neither of these files exist to begin with but `config.yaml` is required
+# Copy `config.yaml.template` to `config.yaml` and edit
+
+# RDS_SNAPSHOT and DATA_SNAPSHOT are only used when restoring a Refinery Platform stack
+# from the EBS and RDS snapshots of a previously created stack
+# If you don't specify these then fresh RDS instances and data volumes will be created
 # RDS_SNAPSHOT: refinery-web-XXX
 # DATA_SNAPSHOT: snap-XXX
 
-# The password for the Django admin account.
-# If not set, an automatically generated one will be placed
-# in `auto-config.yaml`.
+# The password for the Django admin account
 # ADMIN_PASSWORD: "refinery"
 
-
-# Name and URL for the site.
-# You will need to change these.
-SITE_NAME: Refinery
+# Name and URL for the site - change these to match your configuration
+SITE_NAME: "Refinery"
 SITE_URL: "192.168.50.50:8000"
 
-# STACK_NAME is used to name the stack
-# (in the Web UI and other places), and to name instances
+# STACK_NAME is used to name the CloudFormation stacks
 # Constraint: alphanumeric characters, underscores, periods, and colons
 STACK_NAME: "RefineryStack"
-# Prefix for all S3 bucket names
+
+# Prefix for all S3 bucket names (must not be shared among stacks)
 # Constraint: DNS-compliant without periods (".")
 S3_BUCKET_NAME_BASE: "refinery"
 
-# the AWS ARN for the TLS certifcate that you wish to present.
+# Constraint: [\w]+ (must be unique within AWS account)
+COGNITO_IDENTITY_POOL_NAME: "Refinery Platform"
```

Comment:
`\w` stands on its own, and shouldn't be inside `[]`, but apart from that, the regex would not allow the space in the example -- Which is correct?

Would a default value like `Refinery Platform - CHANGE ME` be ok? Or maybe `XXX` as in `RDS_SNAPSHOT`, just so the thing that requires action stands out.

Reply:
Here's the actual description of the field: http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cognito-identitypool.html#cfn-cognito-identitypool-identitypoolname
I suspect [these restrictions](http://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-name.html) apply as well. How would you write a regex for this?

The sample values for `RDS_SNAPSHOT` and `EBS_SNAPSHOT` actually show what those values must look like. On the other hand, `COGNITO_IDENTITY_POOL_NAME` can actually be anything (it doesn't have to start with "Refinery Platform"). Galaxy has a [pretty good example of a config template](https://github.com/galaxyproject/galaxy/blob/dev/config/galaxy.ini.sample) which is what I've tried to follow.

Reply:
`.{1,128}`, though the second page doesn't give space as a valid character? Not worth worrying about. My sense was that this setting is more likely to need changing than others, and it might be good for it to stand out, but it's fine.